flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/CheckpointableOneInputStreamTask.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.tasks;
+
+import org.apache.flink.runtime.checkpoint.CheckpointMetaData;
+import org.apache.flink.runtime.checkpoint.CheckpointMetricsBuilder;
+import org.apache.flink.runtime.checkpoint.CheckpointOptions;
+import org.apache.flink.runtime.execution.Environment;
+
+/** A test stream task that also response to the checkpoint trigger requirement. */
+public class CheckpointableOneInputStreamTask<IN, OUT> extends OneInputStreamTask<IN, OUT> {
```

Comment:
Why don't the tests work anymore with just the regular `OneInputStreamTask`? They also have an implemented `triggerCheckpoint()`.

Reply:
This is due to the logic for this test OneInputStream is different from the formal one in that it would trigger a snapshot (like source stream task) instead of go through the CheckpointBarrierHandler. They are mainly used in some test cases related to taking checkpoint on TM side.

After some rethink I think for these test they would better to directly call `triggerCheckpointOnBarrier` to simulate the case that the alignment has finished. Thus I removed `CheckpointableOneInputStreamTask` and modifies the related tests to call `triggerCheckpointOnBarrier` directly.